src/overlays/actors/ovl_En_Ge3/z_en_ge3.c

```diff
@@ -28,49 +46,386 @@ const ActorInit En_Ge3_InitVars = {
+                                   1200.0f, -1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/EnGe3_Destroy.s")
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A0070.s")
+s32 EnGe3_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnGe3* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A00F8.s")
+    switch (limbIndex) {
+        case GERUDO_RED_LIMB_VEIL:
+        case GERUDO_RED_LIMB_RIGHT_SWORD:
+        case GERUDO_RED_LIMB_LEFT_SWORD:
+            *dList = NULL;
+            return false;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A020C.s")
+        case GERUDO_RED_LIMB_HEAD:
+            rot->x += this->headRot.y;
+            // fallthrough
+        default:
+            // This is required since EnGe3 shares a skeleton with EnKaizoku; it avoids stale colours being used in the
+            // displaylists.
+            OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A024C.s")
+            switch (limbIndex) {
+                case GERUDO_RED_LIMB_NECK:
+                    break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A0350.s")
+                case GERUDO_RED_LIMB_HEAD:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 80, 60, 10, 255);
+                    break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A03AC.s")
+                case GERUDO_RED_LIMB_RIGHT_SWORD:
+                case GERUDO_RED_LIMB_LEFT_SWORD:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 140, 170, 230, 255);
+                    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+                    break;
+                default:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 140, 0, 0, 255);
+                    break;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A03FC.s")
+            CLOSE_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A04D0.s")
+            if (this->animIndex == GERUDO_AVEIL_ANIM_HAND_ON_HIP_WAIT) {
+                // Fidget
+                if ((limbIndex == GERUDO_RED_LIMB_RIGHT_FOREARM) || (limbIndex == GERUDO_RED_LIMB_LEFT_FOREARM) ||
+                    (limbIndex == GERUDO_RED_LIMB_TORSO)) {
+                    s32 temp = limbIndex * 50;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A0820.s")
+                    // required to match
+                    rot->y = rot->y + (s32)(Math_SinS(play->state.frames * (temp + 0x814)) * 200.0f);
+                    rot->z = rot->z + (s32)(Math_CosS(play->state.frames * (temp + 0x940)) * 200.0f);
```

Comment:
Err... what is required? The temp? The Cast?

Reply:
Writing it explicitly as `+` rather than `+=`.

Reply:
Yes, I've never seen it before either, I can scratch it if anyone else wants to try it, but this was what the permuter found to fix the regalloc.